code/modules/mob/living/recoil.dm

```diff
@@ -43,23 +37,57 @@
 	recoil_timer = null
 	update_recoil_cursor(G)
 
-/mob/living/proc/update_recoil_cursor(var/obj/item/weapon/gun/G)
-	G.update_cursor(src)
-	var/bottom = 0
-	switch(recoil)
-		if(0 to 10)
-			;
-		if(10 to 20)
-			bottom = 10
-		if(20 to 30)
-			bottom = 20
-		if(30 to 50)
-			bottom = 30
-		if(50 to MAX_ACCURACY_OFFSET)
-			bottom = 50
-		if(MAX_ACCURACY_OFFSET to INFINITY)
-			bottom = MAX_ACCURACY_OFFSET
-	if(bottom)
-		var/reduction = calc_reduction()
-		if(reduction > 0)
-			recoil_timer = addtimer(CALLBACK(src, .proc/update_recoil_cursor, G), 1 + (recoil - bottom) / reduction)
+/mob/living/proc/update_recoil_cursor()
+	update_cursor()
+	var/reduction = calc_reduction()
+	if(reduction > 0 && recoil > 0)
+		recoil_timer = addtimer(CALLBACK(src, .proc/update_recoil_cursor), 1+round(recoil/10)/reduction)
```

Comment:
Need to fix the math, reduction too slow compared to previous

1+round(recoil/10)/reduction